src/Tasks/Microsoft.NET.Build.Tasks/GenerateGlobalUsings.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class GenerateGlobalUsings : TaskBase
+    {
+        [Required]
+        public ITaskItem[] Usings { get; set; }
+
+        [Required]
+        public ITaskItem OutputFile { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            if (Usings.Length == 0)
+            {
+                return;
+            }
+
+            var usings = Usings.Select(UsingInfo.Read)
+                .OrderBy(static k => k, UsingInfoComparer.Instance)
+                .Distinct(UsingInfoComparer.Instance);
+
+            StringBuilder builder = new();
+            builder.AppendLine("// <auto-generated/>");
+            foreach (var @using in usings)
+            {
+                builder.Append("global using ");
+
+                if (@using.Static)
+                {
+                    builder.Append("static ");
+                }
+
+                if (!string.IsNullOrEmpty(@using.Alias))
+                {
+                    builder.Append(@using.Alias)
+                        .Append(" = ");
+                }
```

Comment:
What if a `Using` item has both `Static` and `Alias` metadata.  This will generate C# code that won't compile, right?  Should we error here in that case?

Reply:
Yeah we should error as `global using static Foo = global::My.Namespace.Value` isn't valid AFAIK.

Reply:
My plan was to let the compiler complain (it produces an error saying that you cannot alias a static using). It felt less complicated to do this in case a newer langversion started supporting this in the future.